feat(discovery): Add a remote_relabel component (#1368)
grafana/alloy: docs/sources/reference/components/discovery/discovery.remote_relabel.md

---
canonical: https://grafana.com/docs/alloy/latest/reference/components/discovery/discovery.remote_relabel/
description: Learn about discovery.remote_relabel
title: discovery.remote_relabel
---

<span class="badge docs-labels__stage docs-labels__item">Experimental</span>

# discovery.remote_relabel

{{< docs/shared lookup="stability/experimental.md" source="alloy" version="<ALLOY_VERSION>" >}}

`discovery.remote_relabel` accepts relabeling rules from a remote control server using Pyroscope's [settings API].
Additionally it is also possible for the control server to request the targets that are received by this component in order to show the effects of the rules at the central control server.

Multiple `discovery.remote_relabel` components can be specified by giving them different labels.

## Usage

```alloy
discovery.relabel "LABEL" {
  targets = TARGET_LIST
  websocket {
    url = "ws://localhost:4040/settings.v1.SettingsService/GetCollectionRules?scope=alloy"
  }
}
```

## Arguments

The following arguments are supported:

| Name      | Type                | Description        | Default | Required |
| --------- | ------------------- | ------------------ | ------- | -------- |

> | `targets` | `list(map(string))` | Targets to relabel |         | yes      |

Reviewer: ```suggestion
| `targets` | `list(map(string))` | Targets to relabel. |         | yes      |
```

## Blocks

The following blocks are supported inside the definition of `pyroscope.write`:

| Hierarchy              | Block          | Description                                               | Required |
| ---------------------- | -------------- | --------------------------------------------------------- | -------- |
| websocket              | [websocket][]  | Control server settingss to.                              | yes      |
| websocket > basic_auth | [basic_auth][] | Configure basic_auth for authenticating to the websocket. | no       |

[websocket]:#websocket-block
[basic_auth]:#basic_auth-block

### websocket block

The `websocket` block describes a single command and control server. Only one `websocket` block can ben provided.

The following arguments are supported:

| Name                  | Type       | Description                                                        | Default   | Required |
| --------------------- | ---------- | ------------------------------------------------------------------ | --------- | -------- |

> | `url`                 | `string`   | Full URL to the websocket server. Needs to start with ws:// wss:// |           | yes      |

Reviewer: ```suggestion
| `url`                 | `string`   | Full URL to the websocket server. Must start with `ws://` or `wss://`. |           | yes      |
```
